app/src/marketplace/flowtype/product-types.js

```diff
@@ -45,12 +45,22 @@ export type Product = {
     pendingChanges?: PendingChanges,
 }
 
+export type ProductSubscriptionId = string
+
+export type ProductSubscriptionIdList = Array<ProductSubscriptionId>
+
 export type ProductSubscription = {
-    address: Address,
+    id: ProductSubscriptionId,
+    address?: Address,
+    user?: string,
```

Comment:
FYI, paid products have the `address` field defined and free ones have the `user` field.